app/uk/gov/hmrc/tai/model/TaxCodeHistory.scala

```diff
@@ -17,12 +17,21 @@
 package uk.gov.hmrc.tai.model
 
 import play.api.libs.functional.syntax._
-import play.api.libs.json.Reads.{minLength, _}
+import play.api.libs.json.Reads._
 import play.api.libs.json.{JsPath, Json, Reads, Writes}
+import uk.gov.hmrc.tai.model.tai.TaxYear
 
 
 case class TaxCodeHistory(nino: String, taxCodeRecords: Seq[TaxCodeRecord]) {
-  def operatedTaxCodeRecords: Seq[TaxCodeRecord] = taxCodeRecords.filter(_.operatedTaxCode)
+  def applicableTaxCodeRecords: Seq[TaxCodeRecord] = inYearTaxCodeRecords(operatedTaxCodeRecords(taxCodeRecords))
```

Comment:
Like how you have combined the two filters here.

Reply:
Could probably also do something like
`removeOutOfYearRecords andThen removeOperatedRecords`